stan/math/prim/mat/prob/multi_normal_lpdf.hpp

```diff
@@ -37,6 +38,15 @@ typename return_type<T_y, T_loc, T_covar>::type multi_normal_lpdf(
   check_ldlt_factor(function, "LDLT_Factor of covariance parameter",
                     ldlt_Sigma);
 
+  size_t number_of_y = length_mvt(y);
+  size_t number_of_mu = length_mvt(mu);
+  if (number_of_y == 0 || number_of_mu == 0)
+    return 0.0;
+  if (number_of_y > 1 && number_of_mu > 1)
+    check_size_match(function, "Number of vectors of the random variable",
```

Comment:
Mind explaining the logic here? Either one of `y` or `mu` has to have two or more elements? 

Is this because `length_mvt()` will return 1 for a single vector?

Wouldn't the sizes be wrong if `y` is legitimately a vector of vectors of length 1 and `mu` is a vector of vectors that's larger?

Reply:
@syclik 

You are correct. The special case is necessary in order to distinguish eigen::vector(1, 2, 3) which should be able to mix with ("is conformable with"?) std::vector(eigen::vector(1, 1, 1), ...). But as it was first written it incorrectly also allows std::vector(eigen::vector(1, 2, 3)) to mix with std::vector(eigen::vector(1, 1, 1), ...).

So really, I need something like:

"contains_vectors" or

```
is_vector<typename std::remove_reference<decltype(x[0])>::type>::value;
```

I've made the necessary changes.

However the upstream tests failed and stopped the build. Can someone take a look?

Reply:
@peterwicksstringfield, I'm really sorry -- I missed this notification. It looks like you've gotten it fixed. I'll take a look.